drivers/sensor/sgp40/sgp40.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021 Leonard Pollak
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_SGP40_SGP40_H_
+#define ZEPHYR_DRIVERS_SENSOR_SGP40_SGP40_H_
+
+#include <device.h>
+#include <kernel.h>
+#include <drivers/gpio.h>
+
+#define SGP40_CMD_MEASURE_RAW	0x260F
+#define SGP40_CMD_MEASURE_TEST	0x280E
+#define SGP40_CMD_HEATER_OFF	0x3615
+
+#define SGP40_TEST_OK		0xD400
+#define SGP40_TEST_FAIL		0x4B00
+
+#define SGP40_RESET_WAIT_MS	10
+#define SGP40_MEASURE_WAIT_MS	30
+#define SGP40_TEST_WAIT_MS	250
+
+/*
+ * CRC parameters were taken from the
+ * "Checksum Calculation" section of the datasheet.
+ */
+#define SGP40_CRC_POLY		0x31
+#define SGP40_CRC_INIT		0xFF
+
+/*
+ * Value range of compensation data parameters
+ */
+#define SGP40_COMP_MIN_RH	0
+#define SGP40_COMP_MAX_RH	100
+#define SGP40_COMP_MIN_T	-45
+#define SGP40_COMP_MAX_T	130
+
+struct sgp40_config {
+	const struct device *bus;
+	uint8_t i2c_addr;
+	bool selftest;
+};
+
+struct sgp40_data {
+	uint16_t raw_sample;
+	int8_t rh_param[3];
+	int8_t t_param[3];
+
+#ifdef CONFIG_PM_DEVICE
+	uint32_t pm_state;
+#endif
+};
+
+static inline int sgp40_clamp(int val, int min, int max) {
+	const int tmp = val < min ? min : val;
+	return tmp > max ? max : tmp;
+}
```

Comment:
https://docs.zephyrproject.org/latest/reference/util/index.html#c.CLAMP

Reply:
i made use of the CLAMP utility. nice!

Reply:
i am now getting warnings while building:
```
include/sys/util.h:188:57: warning: statement with no effect [-Wunused-value]
  188 | #define CLAMP(val, low, high) (((val) <= (low)) ? (low) : MIN(val, high))
      |                               ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
drivers/sensor/sgp40/sgp40.c:66:2: note: in expansion of macro 'CLAMP'
   66 |  CLAMP(rh, SGP40_COMP_MIN_RH, SGP40_COMP_MAX_RH);
      |  ^~~~~
include/sys/util.h:188:57: warning: statement with no effect [-Wunused-value]
```
technically this should not be a problem, right?

Reply:
disregard that last comment... updated again..